### PR TITLE
anonSuggestions

### LINKED ISSUE #15: Anonymous Suggestions posted via mods
An anonymous suggestion command that posts to a mod queue which gets accepted or declined via reactions

### PR DESCRIPTION
**Pull request by Simber1**
Still WIP, needs posting to suggestion channel still. Closes #15

### CODE REVIEW

File: events/messageReactionAdd.js [hunk 1]
```diff
@@ -23,4 +23,34 @@ module.exports = (client, reaction, user) => {
 			}
 		}
 	}
+
+
+	// AnonSuggestion reaction listening
+	const {modBot} = client.config.channelIDs;
```
Comment (NightwingEU) [inline events/messageReactionAdd.js:29]: `const {modBot, suggestions} = client.config.channelIDs;`

File: commands/anonSuggest.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+require('discord.js');
```
Comment (NightwingEU) [inline commands/anonSuggest.js:1]: Unused require

File: commands/anonSuggest.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+module.exports = {
+	name: 'anonsuggest',
+	description: 'Posts an anonymous Suggestion',
+	usage: 'anonsuggest',
```
Comment (NightwingEU) [inline commands/anonSuggest.js:6]: Should make it guildOnly too since it would crash if you dm'd it this

File: events/messageReactionAdd.js [hunk 4]
```diff
@@ -23,4 +23,27 @@ module.exports = (client, reaction, user) => {
...
+	const {suggestions} = client.config.channelIDs;
+
+	if (reaction.message.channel == client.channels.get(modBot)) {
+		if(user.id != client.user.id) {
+			if(reaction.emoji.name == '✔') {
+				let content = reaction.message.content;
```
Comment (NightwingEU) [inline events/messageReactionAdd.js:35]: `let {content} = reaction.message;`

Comment (NightwingEU) [reply events/messageReactionAdd.js:35]: Can also be done via the `lastIndexOf` method of string coupled with `substring()`.

```
const suggestionString = reaction.message.content;
const lastIndex = suggestionString.content.lastIndexOf(' ');
const content = suggestionString.substring(0, lastIndex);
```

File: commands/anonSuggest.js (hunk 2 above)
Comment (Simber1) [reply commands/anonSuggest.js:1]: Pretty sure I took this from your code somewhere

File: events/messageReactionAdd.js (hunk 1 above)
Comment (Simber1) [reply events/messageReactionAdd.js:29]: Fixed

File: commands/anonSuggest.js (hunk 2 above)
Comment (NightwingEU) [reply commands/anonSuggest.js:1]: Nah all requires I've used pull something from it i.e `const {Attachment} = require('discord.js');`

File: events/messageReactionAdd.js [hunk 5]
```diff
@@ -23,4 +23,27 @@ module.exports = (client, reaction, user) => {
 			}
 		}
 	}
+
+
+	// AnonSuggestion reaction listening
+	const {modBot} = client.config.channelIDs;
+	const {suggestions} = client.config.channelIDs;
+
+	if (reaction.message.channel == client.channels.get(modBot)) {
```
Comment (NightwingEU) [inline events/messageReactionAdd.js:32]: triple equals adds type checking and would use it here, and in most cases 
`if (reaction.message.channel === client.channels.get(modBot)) {`

File: events/messageReactionAdd.js (hunk 4 above)
Comment (Simber1) [reply events/messageReactionAdd.js:35]: Done these but better now.

File: commands/anonSuggest.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+	helpMsg: true,
+	execute(message, args) {
+		const {modBot} = message.client.config.channelIDs;
+
+		console.log('running');
```
Comment (NightwingEU) [inline commands/anonSuggest.js:11]: No

Comment (Simber1) [reply commands/anonSuggest.js:11]: Oops, I removed the rest. Missed that one clearly.

File: events/messageReactionAdd.js (hunk 5 above)
Comment (Simber1) [reply events/messageReactionAdd.js:32]: but I like muh abstract equality comparator, I've fixed it now.

File: commands/anonSuggest.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+		const suggestion = args.join(' ');
+
+		message.client.channels.get(modBot).send('"' + suggestion + '": ' + message.author).then(async sent => {
```
Comment (NightwingEU) [inline commands/anonSuggest.js:14]: Put the `.then` on a new line, most other uses of it put it on a new line. Let's see if I can add this in linter too xD

File: commands/anonSuggest.js (hunk 2 above)
Comment (Simber1) [reply commands/anonSuggest.js:1]: I've removed it now

File: commands/anonSuggest.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+module.exports = {
+	name: 'anonsuggest',
+	description: 'Posts an anonymous Suggestion',
+	usage: 'anonsuggest',
+	helpMsg: true,
+	execute(message, args) {
+		const {modBot} = message.client.config.channelIDs;
+		const suggestion = args.join(' ');
+
```
Comment (NightwingEU) [inline commands/anonSuggest.js:11]: I will flame you for this needless empty line

File: commands/anonSuggest.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+		const lastChar = args[0].slice(args[0].length - 1, args[0].length);
+		let number;
+
+		if (!isNaN(lastChar)) {
+			number = args[0];
+		}else {
```
Comment (NightwingEU) [inline commands/anonSuggest.js:17]: spacing bruh, adding the rule into linter now

File: commands/anonSuggest.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+			number = args[0].slice(0, args[0].length - 1);
+		}
+
+		if (isNaN(number)) {
+			message.author.send('Please put a number at the start');
+			if (message.channel instanceof GuildChannel) {
```
Comment (NightwingEU) [inline commands/anonSuggest.js:23]: Don't think you need this here as L35 should pick this up and delete it

File: commands/anonSuggest.js (hunk 8 above)
Comment (Simber1) [reply commands/anonSuggest.js:11]: The others came along later.

File: commands/anonSuggest.js (hunk 9 above)
Comment (Simber1) [reply commands/anonSuggest.js]: Fixed

File: commands/anonSuggest.js (hunk 10 above)
Comment (Simber1) [reply commands/anonSuggest.js:22]: Now it's needed due to return.

**Pull request merged**